cli/pkg/kctrl/cmd/package/available/default_values.go

```diff
@@ -138,28 +137,36 @@ func (s PackageSchema) isKindInt(src interface{}) bool {
 }
 
 func (s PackageSchema) createDefault(structural *structuralschema.Structural, b []bool) []bool {
+	if structural.Type != "object" {
+		return append(b, true)
+	}
 	for _, v := range structural.Properties {
-		// return true if there is a non-nested(not object) with a default value
-		if v.Type != "object" && v.Default.Object != nil {
-			b = append(b, true)
-			return b
+		// return true if there is a non-nested(not object)
+		if v.Type != "object" {
```

Comment:
Are we not able to check if the non-object field is required at this point?

Reply:
There's no required field on the properties. The required field information is available separately on the values schema.

Reply:
This is how it's defined:
```
valuesSchema:
    openAPIv3:
      properties:
        testField:
          description: The test field
          type: string
        requiredField1:
          description: The required field
          type: string
        requiredField2:
          description: Another required field
          type: string
      required:
      - requiredField1
      - requiredField2
```

Reply:
Correct, so should we not emit only the ones that are in the required list? 
(I thought that was the intention)

Reply:
That wasn't the intention of the PR, but do you think we should take that path?
I felt that it would further complicate the code.